src/store.ts

```diff
@@ -306,8 +311,16 @@ export const initializeFirestoreListeners = (onLoad: () => void): (() => void) =
   const subjectColorUnsubscriber = fb.subjectColorsCollection
     .doc(simplifiedUser.email)
     .onSnapshot(snapshot => {
-      const subjectColors = snapshot.data() || {};
-      store.commit('setSubjectColors', subjectColors);
+      const subjectColors = snapshot.data();
+      if (subjectColors == null) {
+        store.commit('setSubjectColors', subjectColors);
```

Comment:
If `subjectColors == null`, doesn't that mean the user has no doc for the `user-subject-colors` collection? Why would we commit null to the store? Would `store.commit('setSubjectColors', {})` be valid?

Reply:
I believe it's causing the bug I took a video of. I deleted my `user-subject-colors` doc and tried to view courses on the requirements bar and add a course but got an error.

https://user-images.githubusercontent.com/54298311/112689263-6c463380-8e50-11eb-8aba-0ac65e7bd3f4.mov

Reply:
fixed